lib/active_model/serializer/association.rb

```diff
@@ -55,7 +55,15 @@ def serializable_hash(adapter_options, adapter_instance)
         association_object = association_serializer && association_serializer.object
         return unless association_object
 
-        serialization = association_serializer.serializable_hash(adapter_options, {}, adapter_instance)
+        # Add this clause to filter fields for associations.
+        # Because attribute adapter doesn't provide fields selection of nested associations.
+        if adapter_options[:include] && adapter_options[:include].is_a?(Hash) && adapter_options[:include][key] && adapter_options[:include][key].is_a?(Hash)
```

Comment:
Why this code checks the truthy of `adapter_options[:include]`? It looks a bit redundant.

```suggestion
        if adapter_options[:include].is_a?(Hash) && adapter_options[:include][key].is_a?(Hash)
```